src/lazy/encoding.rs

```diff
@@ -7,33 +7,64 @@ use crate::lazy::binary::raw::reader::LazyRawBinaryReader;
 use crate::lazy::binary::raw::sequence::{LazyRawBinaryList, LazyRawBinarySExp};
 use crate::lazy::binary::raw::value::LazyRawBinaryValue;
 use crate::lazy::decoder::LazyDecoder;
-use crate::lazy::expanded::macro_evaluator::{ArgumentKind, MacroInvocation, ToArgumentKind};
-use crate::lazy::expanded::EncodingContext;
+use crate::lazy::never::Never;
 use crate::lazy::text::raw::r#struct::LazyRawTextStruct_1_0;
 use crate::lazy::text::raw::reader::LazyRawTextReader_1_0;
 use crate::lazy::text::raw::sequence::{LazyRawTextList_1_0, LazyRawTextSExp_1_0};
 use crate::lazy::text::raw::v1_1::reader::{
     LazyRawTextList_1_1, LazyRawTextReader_1_1, LazyRawTextSExp_1_1, LazyRawTextStruct_1_1,
-    MacroIdRef, RawTextMacroInvocation,
+    RawTextMacroInvocation,
 };
 use crate::lazy::text::value::{
-    LazyRawTextValue_1_0, LazyRawTextValue_1_1, MatchedRawTextValue, RawTextAnnotationsIterator,
+    LazyRawTextValue, LazyRawTextValue_1_0, LazyRawTextValue_1_1, MatchedRawTextValue,
+    RawTextAnnotationsIterator,
 };
-use crate::IonResult;
+use std::fmt::Debug;
+
+/// Marker trait for types that represent an Ion encoding.
+pub trait Encoding: Debug + Copy {
+    fn name() -> &'static str;
+}
```

Comment:
🗺️ I added this partly to have a generic implementation of `Debug` for `RawTextValue<'data, E: TextEncoding>`.